go-controller/cmd/ovn-k8s-cni-overlay/app/helper_windows.go

```diff
@@ -0,0 +1,220 @@
+// +build windows
+
+package app
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"os/exec"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Microsoft/hcsshim"
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types/current"
+
+	"github.com/openvswitch/ovn-kubernetes/go-controller/pkg/config"
+)
+
+// noNameNetNS - This is received for the infra container, in this case we
+// have to create the network endpoint and attach it to the infra container
+// containerNetNS - This is received for all the other containers from
+// the same pod, do not create the network endpoint in thi case
+const (
+	noNameNetNS    = "none"
+	containerNetNS = "container:"
+)
+
+// More details about the above constants can be found in the following PR:
+// https://github.com/kubernetes/kubernetes/pull/51063
+
+// getHNSIdFromConfigOrByGatewayIP returns the HNS Id using the Gateway IP or
+// the config value
+// When the HNS Endpoint is created, it asks for the HNS Id in order to
+// attach it to the desired network. This function finds the HNS Id of the
+// network based on the gatewayIP. If more than one suitable network it's found,
+// return an error asking to give the HNS Network Id in config.
+func getHNSIdFromConfigOrByGatewayIP(gatewayIP string) (string, error) {
+	if config.CNI.WinHNSNetworkID != "" {
+		logrus.Infof("Using HNS Network Id from config: %v", config.CNI.WinHNSNetworkID)
+		return config.CNI.WinHNSNetworkID, nil
+	}
+	hnsNetworkId := ""
+	hnsNetworks, err := hcsshim.HNSListNetworkRequest("GET", "", "")
+	if err != nil {
+		return "", err
+	}
+	for _, hnsNW := range hnsNetworks {
+		for _, hnsNWSubnet := range hnsNW.Subnets {
+			if strings.Compare(gatewayIP, hnsNWSubnet.GatewayAddress) == 0 {
+				if len(hnsNetworkId) == 0 {
+					hnsNetworkId = hnsNW.Id
+				} else {
+					return "", fmt.Errorf("Found more than one network suitable for containers, please specify win-hnsnetwork-id in config")
```

Comment:
Mind making this a multiline message. It makes it easier later on :).

Reply:
done